percy/environment.py

```diff
@@ -104,22 +86,40 @@ def parse(regex):
             else:
                 return None
 
-        # Optional attributes:
-        return {
-            # The only required attribute:
-            'branch': self.branch,
-            # An optional but important attribute:
-            'sha': self.commit_sha or parse(re.compile("COMMIT_SHA:(.*)")),
-
-            # Optional attributes:
-            # If we have the git information, read from those rather than env vars
-            'message': parse(re.compile("COMMIT_MESSAGE:(.*)", flags=re.MULTILINE)),
-            'committed_at': parse(re.compile("COMMITTED_DATE:(.*)")),
-            'author_name': parse(re.compile("AUTHOR_NAME:(.*)")),
-            'author_email': parse(re.compile("AUTHOR_EMAIL:(.*)")),
-            'committer_name': parse(re.compile("COMMITTER_NAME:(.*)")),
-            'committer_email': parse(re.compile("COMMITTER_EMAIL:(.*)")),
-        }
+        if git_data_from_git:
+            return {
+                # The only required attribute:
+                'branch': self.branch,
+                # An optional but important attribute:
+                'sha': self.commit_sha or parse(re.compile("COMMIT_SHA:(.*)")),
+
+                # Optional attributes:
+                # If we have the git information, read from those rather than env vars
+                'message': parse(re.compile("COMMIT_MESSAGE:(.*)", flags=re.MULTILINE)),
+                'committed_at': parse(re.compile("COMMITTED_DATE:(.*)")),
+                'author_name': parse(re.compile("AUTHOR_NAME:(.*)")),
+                'author_email': parse(re.compile("AUTHOR_EMAIL:(.*)")),
+                'committer_name': parse(re.compile("COMMITTER_NAME:(.*)")),
+                'committer_email': parse(re.compile("COMMITTER_EMAIL:(.*)")),
+            }
+
+        # If we don't have git, use environment variables.
+        else:
+            git_data_from_env = {
```

Comment:
could return the object here rather than assigning it perhaps.

Reply:
That's right, I decided to name it to make it clearer what the data was, but I guess the comment does that?